src/libostree/ostree-repo-pull.c

```diff
@@ -2133,6 +2195,65 @@ repo_remote_fetch_summary (OstreeRepo    *self,
   return ret;
 }
 
+static void
+delta_superblock_fetch_on_complete (GObject        *object,
+                         GAsyncResult   *result,
+                         gpointer        user_data)
+{
+  OstreeFetcher *fetcher = (OstreeFetcher *)object;
+  FetchDeltaSuperBlockData *fetch_data = user_data;
+  OtPullData *pull_data = fetch_data->pull_data;
+  GError *local_error = NULL;
+  GError **error = &local_error;
+  GInputStream* input = _ostree_fetcher_stream_uri_finish (fetcher, result, error);
+  GBytes* delta_superblock_data = NULL;
+
+  if (input)
+    {
+      delta_superblock_data = g_input_stream_read_bytes ( input, OSTREE_MAX_METADATA_SIZE, pull_data->cancellable, error);
```

Comment:
What I mean is that if this sets the error, we shouldn't go on and call `delta_superblock_process`.  Although I see that ignores `NULL`...the rest of the async code uses `goto out` and I'd like to continue to match that style.

See for example `content_fetch_on_write_complete()`.  In this case we don't call any further processing and don't increment `pull_data->n_fetched[FETCH_CONTENT]`.